### ISSUE TITLE
Knee of the curve or, worse, GC exception

### ISSUE THREAD

**Issue by FrankC01** [Engelberg/instaparse#92]
Problem:
Built and testing Manchester OWL parsing. I'm using the well known "pizza" ontology and found that (by reducing the number of Class: instances in the file one at a time) there comes a point where the memory and processing time goes off the charts and I am scratching my head about how to resolve. I've tried a few of your performance tips to no avail.

I've narrowed down the problem to the point where I have a file that parses in reasonably linear time (I call this good pizza). With the addition of one more class, performance is noticeably non-linear (I call this bad pizza). Of course trying to parse the full pizza file keeps churning until GC exception time.

Environment: OS X Yosemite
Java: 1.8.0_31-b13
Clojure: 1.6.0
Lein: 2.5.0
instaparse: 1.35

Unfortunately I can't attach any files.

Frank

**Comment by FrankC01** [comment]
Update:

While conducting more tests I noticed that using "pprint" is, well, very piggish on resources. I removed that but were still experiencing the issue. Soo...

I am now in the process of 'pre-chunking' the input and calling parse with the ':start rule' option. I hope to get to meaningful outcome/observations within the next day or two.

However; my working around wouldn't seem to be a solution or even identification of the problem. Is there any guidelines to number of production rules or flattening out recursive productions?

Thanks
Frank

**Comment by FrankC01** [comment]
Update

So, completed chunking up the input and feeding instaparse with targeted start rules. I am able to get through everything without error and am now looking at a **3** second parse versus infinity and exception.

With the 'tweaks' to the grammar I go back and check trying to put the full file through using normal approach of just parse from the first rule, still no luck there.

**Comment by Engelberg** [comment]
You haven't really given me enough information/examples to help troubleshoot, but honestly, I've been so busy, I haven't had much opportunity to help out people who have written for the past few months; even if you had attached examples, I wouldn't be able to look at it right away.  I keep hoping the google group will turn into a forum where people can help each other, but so far, it's mostly me who answers questions.  If you want to attach some specific files and send to the google group, I encourage you to do so, but I can't promise you'll get answers.

In any case, I do appreciate all the updates you've provided.  Although I'm not sure what actions I can immediately take based on this, it is always useful to me to hear what is and is not working for people in real-world use cases.  Thanks for the info.

**Comment by FrankC01** [comment]
No expectations on availability, I figured you were busy and I didn't even think about the google group, I shall subscribe.

Anyway, in my gut I feel it is how I wrote the grammar but I am able to move forward to the 'chunked' approach as *the* method regardless of file size because it is predictable and performant. In the meantime I will post the grammar and the troublesome file but it may be some topic of an academic  discussion when there is idle time.

The :start option is a nice feature as it removed the need for me to create a "chunk" grammar to match each of the file partitions. 

Be well
Frank
